@navikt/core/react/src/overlays/dismissablelayer/DismissableLayer.tsx

```diff
@@ -331,53 +260,153 @@ const DismissableLayerNode = forwardRef<HTMLDivElement, DismissableLayerProps>(
+     * Handles registering `layers` and `layersWithOutsidePointerEventsDisabled`.
      */
-
-    // biome-ignore lint/correctness/useExhaustiveDependencies: Every time the descendants change, we want to update the body pointer events since we might have added or removed a layer.
     useEffect(() => {
-      if (!node || !enabled || !disableOutsidePointerEvents) return;
+      if (!node) {
+        return;
+      }
 
-      if (bodyLockCount === 0) {
-        originalBodyPointerEvents = ownerDoc.body.style.pointerEvents;
-        ownerDoc.body.style.pointerEvents = "none";
+      if (disableOutsidePointerEvents) {
+        if (context.layersWithOutsidePointerEventsDisabled.size === 0) {
+          originalBodyPointerEvents = ownerDoc.body.style.pointerEvents;
+          ownerDoc.body.style.pointerEvents = "none";
+        }
+        context.layersWithOutsidePointerEventsDisabled.add(node);
       }
-      bodyLockCount++;
+      context.layers.add(node);
+      dispatchUpdate();
+
       return () => {
-        if (bodyLockCount === 1) {
+        if (
+          disableOutsidePointerEvents &&
+          context.layersWithOutsidePointerEventsDisabled.size === 1
+        ) {
           ownerDoc.body.style.pointerEvents = originalBodyPointerEvents;
         }
-        bodyLockCount--;
       };
-    }, [node, ownerDoc, disableOutsidePointerEvents, descendants, enabled]);
+    }, [node, disableOutsidePointerEvents, context, ownerDoc]);
+
+    /**
+     * We purposefully prevent combining this effect with the `disableOutsidePointerEvents` effect
+     * because a change to `disableOutsidePointerEvents` would remove this layer from the stack
+     * and add it to the end again so the layering order wouldn't be creation order.
+     * We only want them to be removed from context stacks when unmounted.
+     */
+    useEffect(() => {
+      return () => {
+        if (!node) {
+          return;
+        }
+
+        context.layers.delete(node);
+        context.layersWithOutsidePointerEventsDisabled.delete(node);
+        dispatchUpdate();
+      };
+    }, [node, context]);
+
+    const parentBranchedLayer = useContext(BranchedLayerContext);
 
     /**
-     * To make sure pointerEvents are enabled for all parents and siblings when the layer is removed from the DOM
+     * Handles registering and unregistering branched layers.
+     * When this layer has a parent, we register it as a child of the parent.
      */
-    // biome-ignore lint/correctness/useExhaustiveDependencies: We explicitly want to run this on unmount, including every time the node updates to make sure we don't lock the application behind pointer-events: none.
     useEffect(() => {
-      return () => descendants.values().forEach((x) => x.forceUpdate());
-    }, [descendants, node]);
+      if (!node || !parentBranchedLayer || node === parentBranchedLayer) {
+        return;
+      }
+
+      // Get or create the Set of children for this parent
```

Comment:
```suggestion
      // Create the Set of children for this parent, if it does not already exist
```
We might not need this comment at all though

Reply:
True, just removed it 👍